packages/protoplugin-test/src/custom-options.test.ts

```diff
@@ -0,0 +1,242 @@
+        1234
+      );
+    });
+    test("field options", () => {
+      const msg = descriptorSet.messages.get("example.FooMessage");
+      assert(msg);
+      for (const member of msg.members) {
+        switch (member.kind) {
+          case "oneof":
+            expect(
+              findCustomScalarOption(member, 50003, ScalarType.INT64)
+            ).toEqual(BigInt(42));
+            break;
+          default: {
+            const val = findCustomScalarOption(member, 50002, ScalarType.FLOAT);
+            if (member.name === "foo") {
+              expect(val).toEqual(4.5);
+            } else {
+              expect(val).toBeUndefined();
+            }
+            break;
+          }
+        }
+      }
+    });
+    test("service options", () => {
+      const service = descriptorSet.services.get("example.FooService");
+      assert(service);
+      expect(findCustomEnumOption(service, 50006)).toEqual(1);
+    });
+    test("method options", () => {
+      const service = descriptorSet.services.get("example.FooService");
+      assert(service);
+      for (const method of service.methods) {
+        const option = findCustomMessageOption(method, 50007, FooMessage);
+        expect(option?.foo).toEqual(567);
+        expect(option?.bar).toEqual("Some string");
+        expect(option?.qux.case).toEqual("quux");
+        expect(option?.qux.value).toEqual("Oneof string");
+        expect(option?.many).toEqual(["a", "b", "c"]);
+        expect(option?.mapping).toEqual({ testKey: "testVal" });
+      }
+    });
+  });
+  describe("all methods return undefined when option not found", function () {
+    test("file options", () => {
+      for (const file of descriptorSet.files) {
+        expect(
+          findCustomScalarOption(file, 99999, ScalarType.STRING)
+        ).toBeUndefined();
+      }
+    });
+    test("enum options", () => {
+      for (const file of descriptorSet.files) {
+        for (const enumeration of file.enums) {
+          expect(
+            findCustomScalarOption(enumeration, 99999, ScalarType.BOOL)
+          ).toBeUndefined();
+        }
+      }
+    });
+    test("enum value options", () => {
+      for (const file of descriptorSet.files) {
+        for (const enumeration of file.enums) {
+          for (const enumValue of enumeration.values) {
+            expect(
+              findCustomScalarOption(enumValue, 99999, ScalarType.UINT32)
+            ).toBeUndefined();
+          }
+        }
+      }
+    });
+    test("message options", () => {
+      for (const file of descriptorSet.files) {
+        for (const message of file.messages) {
+          expect(
+            findCustomScalarOption(message, 99999, ScalarType.INT32)
+          ).toBeUndefined();
+        }
+      }
+    });
+    test("field options", () => {
+      for (const file of descriptorSet.files) {
+        for (const message of file.messages) {
+          for (const member of message.members) {
```

Comment:
This should also look up by name. Otherwise, renaming a field in the proto source will skip the assertions in this test without us noticing. 

For example:

```ts
      const m = descriptorSet.messages.get("example.FooService")?.members.find(m => m.name === "qux");
      assert(m);
      assert(m.kind === "oneof");
      findCustomScalarOption(m, 99999, ScalarType.INT64)
```